trace/config.go

```diff
@@ -189,16 +189,42 @@ type SpanOption interface {
 	SpanEndOption
 }
 
-// SpanStartEventOption are options that can be used at the start of a span, or with an event.
-type SpanStartEventOption interface {
+// SpanStartEventErrorOption are options that can be used at the start of a span, or with an event, or when recording
+// an error.
+type SpanStartEventErrorOption interface {
 	SpanStartOption
 	EventOption
+	ErrorOption
 }
 
-// SpanEndEventOption are options that can be used at the end of a span, or with an event.
-type SpanEndEventOption interface {
+// SpanEndEventErrorOption are options that can be used at the end of a span, or with an event, or when recording an
+// error.
+type SpanEndEventErrorOption interface {
 	SpanEndOption
 	EventOption
+	ErrorOption
+}
+
+type ErrorConfig struct {
+	setErrorStatus bool
+}
+
+func (cfg *ErrorConfig) SetErrorStatus() bool {
```

Comment:
Calling this `SetErrorStatus` kind of hints at a setter, which this isn't. How about naming the variable `hasErrorStatus`, and this method `HasErrorStatus`?

Reply:
Agreed, will fix